### PR TITLE
Purify Chamber status fix

### PR DESCRIPTION
**Pull request by RegisCoaxans**
## Description
The area status computation for the Purify Chamber had to change to make sure it always returns a valid status.

## Motivation and Context
When it was unlocked but not fulfilled, its status was computed as undefined, leading to no map overlay for Relic Stone and Agate Village.
See the [bug report](https://discord.com/channels/450412847017754644/1145408320488882186) on the Discord server.

## How Has This Been Tested?
Opened the game, unlocked Purify Chamber, and checked the map overlays computed correctly.

## Types of changes
- Bug fix

### CODE REVIEW

**Comment by Farboo171** [comment]
Doing a test, I'm getting the Uncaught Pokemon overlay when I have the following:

- some shadow pokemon uncaught, but not in my party
- all shadow pokemon in my party purified
- all shadow pokemon in Relic Cave are captured

Also getting the Uncaught Pokemon overlay when:

- some shadow pokemon uncaught, but not in my party
- one shadow pokemon in my party is not yet purified
- I do not have sufficient flow to purify

and:

- some shadow pokemon uncaught, but not in my party
- one shadow pokemon in my party is not yet purified
- I *do* have sufficient flow to purify

Basically, I can't see anything other than the Uncaught Pokemon filter, even when it should be filter-less

**Comment by RegisCoaxans** [comment]
I forgot part of the code... Should work as intended now.
